netty/src/main/java/io/grpc/netty/AbstractNettyHandler.java

```diff
@@ -108,4 +113,95 @@ private void sendInitialConnectionWindow() throws Http2Exception {
       ctx.flush();
     }
   }
+
+  /**
+   * Class for handling flow control pinging and flow control window updates as necessary
+   */
+  protected class FlowControlPinger {
+
+    private static final int BDP_MEASUREMENT_PING = 1234;
+    private static final int MAX_WINDOW_SIZE = 8 * 1024 * 1024;
+
+    private int pingCount;
+    private int pingReturn;
+    private boolean pinging;
+    private int dataSizeSincePing;
+    private ByteBuf payloadBuff;
```

Comment:
(and honestly, I don't care too much at this point whether we reuse a buffer or create a new one each time)